### PR TITLE
Don't add cache dependency in Liquibase if cache is disabled

### LINKED ISSUE #7892: Maven build failure - probably with no cache
##### **Overview of the issue**

I synced my test project with latest master and noted below stack trace.

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 1 of constructor in com.mycompany.myapp.config.LiquibaseConfiguration required a bean of type 'org.springframework.cache.CacheManager' that could not be found.
        - Bean method 'cacheManager' not loaded because @ConditionalOnClass did not find required class 'com.github.benmanes.caffeine.cache.Caffeine'
        - Bean method 'cacheManager' not loaded because @ConditionalOnClass did not find required class 'net.sf.ehcache.Cache'
        - Bean method 'cacheManager' not loaded because @ConditionalOnClass did not find required class 'javax.cache.Caching'

Action:

Consider revisiting the conditions above or defining a bean of type 'org.springframework.cache.CacheManager' in your configuration.

##### **Suggest a Fix**

I guess it's related to no cache selection during generation, and with Liquibase code refactoring it's kind of required, but, I may be wrong.

##### **JHipster configuration**

{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.0.0-beta.1",
    "baseName": "rgtw3",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "<secret>",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "es"
    ]
  }
}

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by vishal423**
Closes #7892 
- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by jdubois** [comment]
Thanks, that's probably my fault, I'll have a look

File: generators/server/templates/src/main/java/package/config/LiquibaseConfiguration.java.ejs [hunk 1]
```diff
@@ -24,7 +24,11 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.autoconfigure.liquibase.LiquibaseProperties;
+<%_ if (databaseType === 'sql') { _%>
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/config/LiquibaseConfiguration.java.ejs]: there's always going to be SQL here, it only works for SQL databases

**Comment by jdubois** [comment]
This should be good, let's wait for Travis to pass but this should be merged

File: generators/server/templates/src/main/java/package/config/LiquibaseConfiguration.java.ejs (hunk 1 above)
Comment (vishal423) [reply generators/server/templates/src/main/java/package/config/LiquibaseConfiguration.java.ejs]: yeah, I just took old code and didn't paid much attention here. I noted that you removed, so, we should be good to go

**Pull request merged**